### PR TITLE
Clarify S3 bucket name restrictions

### PR DESCRIPTION
**Pull request by mikemartin1090**
I ran into an issue today with creating an s3 bucket. Terraform outputs a non-descriptive error about an invalid bucket name. I would have quickly caught this if I had a reference to the rules. Hoping this will help some others that are looking for a quick bucket creation.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/r/s3_bucket.html.markdown [hunk 1]
```diff
@@ -337,8 +337,8 @@ resource "aws_s3_bucket" "bucket" {
 
 The following arguments are supported:
 
-* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be less than or equal to 63 characters in length.
-* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be less than or equal to 37 characters in length.
+* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be lowercase and less than or equal to 63 characters in length. [A full list of bucket naming rules can be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:340]:
```suggestion
* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be lowercase and less than or equal to 63 characters in length. [A full list of bucket naming rules may be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```

File: website/docs/r/s3_bucket.html.markdown [hunk 2]
```diff
@@ -337,8 +337,8 @@ resource "aws_s3_bucket" "bucket" {
...
+* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be lowercase and less than or equal to 37 characters in length. [A full list of bucket naming rules can be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:341]:
```suggestion
* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be lowercase and less than or equal to 37 characters in length. [A full list of bucket naming rules may be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```

**Review by justinretzolk — APPROVED** [review]
Hey @mikemartin1090 👋 Thank you for your contribution! I made two _extremely_ small suggestions, but overall this looks good to me, so I've approved as well. After you've taken a look over the suggestions, let me know and I'll get this merged in.

**Comment by mikemartin1090** [comment]
I appreciate the suggestions and have incorporated them. Thank you!

File: website/docs/r/s3_bucket.html.markdown [hunk 3]
```diff
@@ -337,8 +337,8 @@ resource "aws_s3_bucket" "bucket" {
 
 The following arguments are supported:
 
-* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be less than or equal to 63 characters in length.
-* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be less than or equal to 37 characters in length.
+* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be lowercase and less than or equal to 63 characters in length. [A full list of bucket naming rules may be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:340]:
```suggestion
* `bucket` - (Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name. Must be lowercase and less than or equal to 63 characters in length. [A full list of bucket naming rules may be found here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).
```

File: website/docs/r/s3_bucket.html.markdown [hunk 4]
```diff
@@ -337,8 +337,8 @@ resource "aws_s3_bucket" "bucket" {
...
+* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be lowercase and less than or equal to 37 characters in length. [A full list of bucket naming rules may be found here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:341]:
```suggestion
* `bucket_prefix` - (Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with `bucket`. Must be lowercase and less than or equal to 37 characters in length. [A full list of bucket naming rules may be found here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).
```

**Review by justinretzolk — APPROVED** [review]
Apologies for the back and forth here, @mikemartin1090 - I overlooked one additional thing on the first pass. I'd generally expect to see the period after the link rather than as a part of it. After this, I'll get this merged in. Thank you again for your contribution!

**Comment by justinretzolk** [comment]
Thank you again for your contribution @mikemartin1090! 🎉

**Pull request merged**